script.js

```diff
@@ -1,12 +1,333 @@
+    expandMoreContent.appendChild(showSummary);
+    // const expandMoreHolder = document.createElement("section");
+    // expandMoreHolder.className = "expandMoreHolder";
+    // description.appendChild(expandMoreHolder);
+    // const spanEl = document.createElement("span");
+    // spanEl.setAttribute("data-hidetext", "show less...");
+    // spanEl.dataset = "expand-more";
+    // spanEl.setAttribute("data-showtext", "show more...");
+    // spanEl.setAttribute("data-target", "showMoreContent");
+    // spanEl.className = "btn-expand-more";
+    // spanEl.innerHTML = "...read more";
+    // expandMoreHolder.appendChild(spanEl);
+    // document.addEventListener("DOMContentLoaded", () => {
+    //   const expandsMore = document.querySelectorAll("[expand-more]");
+    //   function expand() {
+    //     const showContent = document.getElementById(this.dataset.target);
+    //     if (showContent.classList.contains("expand-active")) {
+    //       this.innerHTML = this.dataset.showtext;
+    //     } else {
+    //       this.innerHTML = this.dataset.hidetext;
+    //       showContent.classList.toggle('expand-active')
+    //     }
+    //   }
+    //   expandsMore.forEach((expandsMore) => {
+    //     expandsMore.addEventListener("click", expand);
+    //   });
+    //});
+  });
 }
 
+let showUrl;
+
+let selectShowEl = document.getElementById("select-show");
+selectShowEl.addEventListener("change", (e) => {
+  let selectedTerm = e.target.value;
+  let showID;
+  loadShowList(selectedTerm);
+  let selectedShow = allShows.filter((show) => show.name === selectedTerm);
+  showID = selectedShow[0].id;
+  loadShowList(showID);
+
+  containerEl.innerHTML = "";
+  selectShowEl.style.display = "none";
+  searchShowEl.style.display = "none";
+  selectEl.style.display = "block";
+  searchEl.style.display = "block";
+  resetEl.style.display = "block";
+  homeEl.style.display = "block";
+  document.querySelector(".search-icon").style.display = "block";
+  displayCountEl.innerHTML = `Displaying ${selectedShow.length}/${allShows.length} show(s)`;
+});
+
+// search show
+
+let searchShowEl = document.getElementById("search-show");
+searchShowEl.addEventListener("keyup", (e) => {
+  const searchTerm = e.target.value.toLowerCase();
+
+  console.log(searchTerm);
+  const searchedShow = allShows.filter((show) => {
+    return (
+      show.name.toLowerCase().includes(searchTerm) ||
+      show.genres.includes(searchTerm) ||
+      show.summary.toLowerCase().includes(searchTerm)
+    );
+  });
+  // console.log(searchedShow);
+  containerEl.innerHTML = "";
+  selectShowEl.innerHTML = "";
+  let selectOption = document.createElement("option");
+  selectOption.innerHTML = "---select show---";
+  selectShowEl.appendChild(selectOption);
+  displayAllShows(searchedShow);
+  displayCountEl.innerHTML = `Displaying ${searchedShow.length}/${allShows.length} show(s)`;
+});
+
+// using the API fetch
+
+const setup = async () => {
```

Comment:
Well written asynchronous setup function!
Moving it up to somewhere like line 7 would be a good idea as it is the entry point to your application.